DXMainClient/Domain/Multiplayer/MapLoader.cs

```diff
@@ -55,6 +88,121 @@ public void LoadMapsAsync()
             thread.Start();
         }
 
+        /// <summary>
+        /// Start the file watcher for the custom map directory.
+        ///
+        /// This will refresh the game modes and map lists when a change is detected. 
+        /// </summary>
+        public void StartCustomMapFileWatcher()
+        {
+            customMapFileWatcher = new FileSystemWatcher($"{ProgramConstants.GamePath}{CustomMapsDirectory}");
```

Comment:
```suggestion
            customMapFileWatcher = new FileSystemWatcher(SafePath.CombineDirectoryPath(ProgramConstants.GamePath, CustomMapsDirectory));
```

Concatenating string paths won't work cross-platform. Always use SafePath when evaluating file or folder paths.

Reply:
Thanks for the review. I think I found everywhere where I was concatenating paths and replaced them.